### ISSUE TITLE
24.3.1.2 SetViewValue - Discrepancy between spec and implementations?

### ISSUE THREAD

**Issue by raulsebastianmihaila** [tc39/ecma262#1706]
Not sure if this is a spec bug but it seems to me that there is a discrepancy between Chrome/Firefox and the spec. Am I missing something?
Step 5 of [24.3.1.2 SetViewValue](https://tc39.es/ecma262/#sec-setviewvalue) converts the value into an integer, which means that `NaN` is converted into `0`. However in Chrome and Firefox `DataView.prototype.setFloat64` called with value `0` doesn't have the same effect as called with value `NaN`:

```js
var b = new ArrayBuffer(8);
var a = new Uint8Array(b);
var d = new DataView(b);
d.setFloat64(0, NaN);
a; // [ 127, 248, 0, 0, 0, 0, 0, 0 ]
d.setFloat64(0, 0)
a; // [ 0, 0, 0, 0, 0, 0, 0, 0 ]
```

**Comment by devsnek** [comment]
It should be calling `ToNumber` there, this happened in the bigint addition: https://github.com/tc39/ecma262/commit/fd3a2604fd00e45e524553614c75aef77bc46a80#diff-3540caefa502006d8a33cb1385720803R36845

**Comment by ljharb** [comment]
Fixed in #1708.
